File: ebean-test/src/test/java/org/tests/model/join/initfields/OrderItem.java
package org.tests.model.join.initfields;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name="join_initfields_order_item")
public class OrderItem {
  @Id
  int id;

  @ManyToOne
  Order order;

  public OrderItem(Order order) {
    this.order = order;
  }
}

File: tests/test-kotlin/src/main/kotlin/org/example/OrderItem.kt
package org.example.order

import java.math.BigDecimal
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne

@Entity
class OrderItem(
  order: Order,
  name: String,
  price: BigDecimal,
  amount: Int = 1,
  description: String? = null,
) {
  @Id
  val id: Long = 0

  @ManyToOne
  val order: Order = order

  val name: String = name

  val description: String? = description

  val amount: Int = amount

  val price: BigDecimal = price
}
